### PR TITLE
Sig Scalability 2021 annual report

### LINKED ISSUE #6378: 2021 Annual Report: SIG Scalability
Chairs: @marseel @shyamjvs 
Liaison: @mrbobbytables

Actions for the chair/organizer of the community group:
- [ ] Consult your community group to complete draft of report
- [ ] If needed, consult with Steering Committee liaison on any private issues or concerns that arrise while completing report
- [ ] Submit PR to git.k8s.io/community repo with copy of report, assigning your Steering Committee liaison for review
- [ ] Steering Committee reviews report, and may make comments/ask questions
- [ ] Once all comments are addressed, Steering Committee will approve report to merge
- [ ] If needed, any follow up items may be brought to the "Chairs, Tech Leads, and Organizers" meeting in April

Once all the above items are complete, this issue may be `/close`'d

Key dates:
- Initial PR to communtiy repo should be opened by February 14, 2022
- PR should be reviewed and merged by March 1st, 2022

More detailed information on the annual reports process is available [here](https://git.k8s.io/community/committee-steering/governance/annual-reports.md).

Your group's report template is available here: https://git.k8s.io/community/sig-scalability/annual-report-2021.md

If you have any questions or concerns about this process, you may reach the Steering Committee via the following methods:
- Slack/E-mail your liaison directly
- The public #steering-committee channel in Slack
- The public mailing list steering@kubernetes.io
- The private mailing list steering-private@kubernetes.io for any private or sensitive issues.

/sig scalability
/committee steering

### PR DESCRIPTION
**Pull request by marseel**
**Which issue(s) this PR fixes**:

Fixes #6378

Initial draft for sig-scalability annual report.

### CODE REVIEW

File: sig-scalability/README.md [hunk 1]
```diff
@@ -161,7 +161,10 @@ more generalized benchmarking scenarios.
 
 Ensuring that all tests necessary to validate Kubernetes scalability and
 performance exists (ideally by providing easy-to-use frameworks and working
-with SIGs to provide them), having environment and resources to run them.
+with SIGs to provide them), having environment and resources to run them:
```
Comment (wojtek-t) [inline sig-scalability/README.md:164]: move those changes to a separate PR

File: sig-scalability/annual-report-2021.md [hunk 2]
```diff
@@ -4,79 +4,62 @@
...
-   -
-
-3. KEP work in 2021 (1.x, 1.y, 1.z):
-
-<!--
-In future, this will be generated from kubernetes/enhancements kep.yaml files
-1. with SIG as owning-sig or in participating-sigs
-2. listing 1.x, 1.y, or 1.z in milestones or in latest-milestone
--->
+   - TBD
```
Comment (wojtek-t) [inline sig-scalability/annual-report-2021.md:11]: All testing infra improvements

File: sig-scalability/annual-report-2021.md [hunk 3]
```diff
@@ -4,79 +4,62 @@
...
 
+3. KEP work in 2021:
    - Stable
-     - [$kep-number - $title](https://git.k8s.io/community/$link/README.md) - $milestone.stable
-     - [$kep-number - $title](https://git.k8s.io/community/$link/README.md) - $milestone.stable
+     - [1040 - Priority and Fairness for API Server Requests](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1040-priority-and-fairness/README.md) - 1.23
```
Comment (wojtek-t) [inline sig-scalability/annual-report-2021.md:15]: P&F is not stable - it's in beta

File: sig-scalability/annual-report-2021.md [hunk 4]
```diff
@@ -4,79 +4,62 @@
...
 
 ## Project health
 
 1. What areas and/or subprojects does your group need the most help with?
    Any areas with 2 or fewer OWNERs? (link to more details)
 
-   -
-   -
-   -
+   - TBD
```
Comment (wojtek-t) [inline sig-scalability/annual-report-2021.md:30]: It didn't visibly change since last year.

Some answers may be mostly copied from previous year:
https://github.com/kubernetes/community/blob/master/sig-scalability/annual-report-2020.md

File: sig-scalability/README.md (hunk 1 above)
Comment (marseel) [reply sig-scalability/README.md]: Done https://github.com/kubernetes/community/pull/6503

File: sig-scalability/annual-report-2021.md (hunk 3 above)
Comment (marseel) [reply sig-scalability/annual-report-2021.md:15]: Done, we should probably update this too: https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1040-priority-and-fairness/kep.yaml

**Comment by marseel** [comment]
The only thing that is left is adding CONTRIBUTING.md
Please take another look @wojtek-t

**Comment by wojtek-t** [comment]
This looks reasonable to me.

@mrbobbytables - WDYT?

File: sig-scalability/annual-report-2021.md [hunk 5]
```diff
@@ -4,79 +4,66 @@
...
-   -
-   -
-   -
+   - We care mostly about metrics for our main repository [perf-tests]. We measure them through [devstats]. We keep track of number of reviewers, opened issues and PRs merged.
 
 3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
    to activities or programs that provide useful context or allow easy participation?
 
-   -
+   - Yes, [CONTRIBUTING.md] points to maintained list of issues with label `good first issue`.
```
Comment (mrbobbytables) [inline sig-scalability/annual-report-2021.md:43]: I don't see a CONTRIBUTING.md file in the sig scalability dir :x is this a follow up item?

https://github.com/kubernetes/community/tree/master/sig-scalability

File: sig-scalability/annual-report-2021.md [hunk 6]
```diff
@@ -4,79 +4,66 @@
...
 4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
    does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?
 
-   -
+   - We don't have any special requirements for reviewers/approvers.
 
 5. Does the group have contributors from multiple companies/affiliations?
 
-   -
+   - Yes, we have contributors from multiple companies.
```
Comment (mrbobbytables) [inline sig-scalability/annual-report-2021.md:52]:
```suggestion
   - Yes, we have contributors from [multiple companies](https://k8s.devstats.cncf.io/d/8/company-statistics-by-repository-group?orgId=1&var-period=d7&var-metric=contributions&var-repogroup_name=SIG%20Scalability&var-repo_name=kubernetes%2Fkubernetes&var-companies=All&from=1609477200000&to=1640926800000).
```
Not a blocker by any means, but this will link out to devstats showing the different companies for 2021 :)

File: sig-scalability/annual-report-2021.md [hunk 7]
```diff
@@ -4,79 +4,66 @@
...
    to activities or programs that provide useful context or allow easy participation?
 
-   -
+   - Yes, [CONTRIBUTING.md] points to maintained list of issues with label `good first issue`.
 
 4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
    does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?
 
-   -
+   - We don't have any special requirements for reviewers/approvers.
```
Comment (mrbobbytables) [inline sig-scalability/annual-report-2021.md:48]: I know you mentioned this above, but it might be worth calling out again down here that core areas of sig-scalability require deep and wide understanding of the project.

For the CONTRIBUTING.md doc If someone did want to start down the track of growing in sig-scalability, what would be a good entry point for them? is there any background knowledge or experience that might be useful? like what would you put in a job posting if looking to hire someone to help.

**Review by mrbobbytables — COMMENTED** [review]
Thanks for updating, just have 1 question and 2 suggestions :)

File: sig-scalability/annual-report-2021.md (hunk 5 above)
Comment (marseel) [reply sig-scalability/annual-report-2021.md:43]: Yes, this is follow up item that I am working on right now, see my comment above :)

File: sig-scalability/annual-report-2021.md (hunk 7 above)
Comment (marseel) [reply sig-scalability/annual-report-2021.md:48]: Added it here.

Sounds good, I will add "job posting" like description to CONTRIBUTING.md

**Comment by mrbobbytables** [comment]
> The only thing that is left is adding CONTRIBUTING.md

gah, sorry - I completely missed that comment 🤦

**Comment by marseel** [comment]
I've opened PR for CONTRIBUTING.md: https://github.com/kubernetes/community/pull/6516

**Comment by mrbobbytables** [comment]
lgtm! next step would be to fire it off to your list with a lazy consensus period to merge 👍

**Comment by wojtek-t** [comment]
/lgtm
/approve

/hold
Hold for lazy consensus until Thu EOD PST.

**Comment by wojtek-t** [comment]
Lazy consensus email: https://groups.google.com/g/kubernetes-sig-scale/c/DAjw-JzyKBs
